routes/activities_products.js

```diff
@@ -0,0 +1,38 @@
+const express = require('express');
+const connection = require('../conf');
+const router = express.Router({ mergeParams: true });
+
+router.get('/', (req, res) => {
+  connection.query('SELECT * from Activities_Products', (err, results) => {
+    if (err) {
+      return (
+        res.status(500).json({ message: 'Error server' })
+      )
+    }
+    res.json(results);
+  });
+});
+
+router.get('/:id', (req, res) => {
+  const { id } = req.params
+  if (isNaN(id)) {
```

Comment:
NaN is a special number value: the string 'foo' is not a number, but it isn't a NaN either.

you may follow this pattern instead: https://github.com/rocambille/back-to-mvc/blob/91b743e6d6e999aba7ee9792c26ef8103ba21ac3/index.js#L16

Reply:
try removing this if